tests/ui/restrictions/mut-restriction-construction.stderr

```diff
@@ -0,0 +1,24 @@
+error: `Foo` cannot be constructed using a struct expression outside `foo`
+  --> $DIR/mut-restriction-construction.rs:14:15
+   |
+LL |         pub mut(self) alpha: u8,
+   |             --------- mutability restricted here
+...
+LL |     let foo = foo::Foo { alpha: 0 };
+   |               ^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: a struct containing fields with a mutability restriction cannot be constructed using a struct expression
```

Comment:
Huh, this is kind of surprising to me, but I guess that's a question more for the RFC than this PR.

Reply:
It is admittedly an unresolved question. The reason I favor it is that it's permitting an arbitrary value to be assigned to the field, which is most likely what is trying to be avoided by restricting mutation.

The message is quite verbose, though, and I've been unable to think of anything more concise. If you or anyone else has suggestions, I'm all ears.

Reply:
I think the error message is okay. It didn't strike me as overly long, but I found it clear and easy to follow.

Your reasoning makes sense to me. Especially if you consider that you could do the following:

```rust
let foo = foo::Foo::new();
let foo = foo::Foo { alpha: 0, ..foo };
```

Very likely you'd be able to violate any invariants the `foo` module is trying to uphold, which seems like it kind of defeats the purpose of this feature.